.github/workflows/tests.yml

```diff
@@ -40,10 +40,39 @@ jobs:
         if: github.event_name == 'pull_request'
         id: formulae-detect
 
+  timeout_labels:
+    if: github.event_name == 'pull_request' && github.repository == 'Homebrew/homebrew-core'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Count pull requests with the `CI-long-timeout` label
+        if: contains(github.event.pull_request.labels.*.name, 'CI-long-timeout')
```

Comment:
Except this doesn't do what I thought it would do, which is just run this check only when the PR itself has the `CI-long-timeout` label.